tests/regression/22-partitioned_arrays/10-array_octagon.c

```diff
@@ -96,8 +96,10 @@ void example1(void) {
   assert(a[z] == 1); //UNKNOWN
   assert(a[z] == 2); //UNKNOWN
   assert(z >= 0);
-  assert(z <= j);
-  assert(a[z] == 0); //FAIL
+  assert(z <= j); //UNKNOWN
+
+  // Used to work with the custom octagon analysis, does not work with apron octagons
+  assert(a[z] != 0); //UNKNOWN
```

Comment:
The thing not working on arrays here is a direct result of the octagon analysis not knowing that `z <=j ` holds, which is a bit strange because that should be known.

Reply:
With `--set sem.int.signed_overflow assume_none` it passes, to overflows are somehow the reason. I think it doesn't have good bounds for `j`, so `j - 1` might underflow and would by default be set to top.
I suppose we can just add that parameter to the tests where octagon analysis was previously more precise and see if that makes them all pass. I would think it should be able to.

Reply:
> I suppose we can just add that parameter to the tests where octagon analysis was previously more precise and see if that makes them all pass. I would think it should be able to.

I changed the tests accordingly.